Changed the description of the 'application lock mode for reindex pro… (#9048)
magento/devdocs: src/guides/v2.4/extension-dev-guide/indexing.md

### Using application lock mode for reindex processes

> Starting with 2.4.3, you can enable `use_application_lock` mode for reindexing through the use of environment variables, or by configuring the `app/etc/env.php` file like so:

Reviewer: ```suggestion
Starting with 2.4.3, you can enable `use_application_lock` mode for reindexing through the use of environment variables or by configuring the `app/etc/env.php` file like so:
```

];
```

In case of a failure during the reindexing of a certain indexer, this mode will return a more accurate status of the indexer.
The status can be obtained from the indexer grid in the Magento Admin or through the `bin/magento indexer:status` command in the CLI.

{:.bs-callout-warning}

> The status values in the `indexer_state` or `mview_state` database tables may not be the same as what you are seeing because they sometimes do not get updated when an indexer fails.

Reviewer: ```suggestion
The status values in the `indexer_state` or `mview_state` database tables may not be the same as what is observed, because they sometimes do not get updated when an indexer fails.
```

> An additional benefit of this mode is that the Magento application will internally also see a more accurate status of the indexers and if an indexer failed, Magento will now see this and the cronjob will pick up the indexer again to try it again.

Reviewer: ```suggestion
An additional benefit of this mode is that the Magento application will, internally, also see a more accurate status of the indexers and if an indexer failed, Magento will now see this and the cronjob will pick up the indexer to try it again.
```

> Without this mode, you needed to manually reset the indexer when it failed. With this mode enabled, this is no longer the case anymore.

Reviewer: ```suggestion
Without this mode, it was necessary to manually reset the indexer when it failed. With this mode enabled, this is no longer the case.
```
